src/ERC4973.sol

```diff
@@ -77,4 +79,18 @@ abstract contract ERC4973 is ERC165, IERC721Metadata, IERC4973 {
 
     emit Revoke(owner, tokenId);
   }
+
+  function _mintWithPermission(
+    address from,
+    address to,
+    uint256 tokenId,
+    string memory uri,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  ) internal virtual returns (uint256) {
+    require(!_isPermittedToMint(from, to, uri, v, r, s), "_mintWithPermission: unauthorized caller");
```

Comment:
I think you may want to change this line to `require(_isPermittedToMint)` so that it's reads:

"It is `required` that `isPermittedToMint` is true.